src/utils/performance_optimizer.py

```
@@ -9,203 +1,65 @@
+        try:
+            yield metrics
+        finally:
+            end_time = time.time()
+            end_memory = self.get_memory_usage()
 
-        avg_latency = sum(self._latency_history) / len(self._latency_history)
-        error_rate = len(self._error_history) / len(self._latency_history)
+            metrics.update({
+                'duration_ms': (end_time - start_time) * 1000,
+                'memory_usage': end_memory,
+                'memory_delta': end_memory - start_memory
```

Ruff (0.8.2)

Trailing comma missing

Add trailing comma

(COM812)